autoapi/documenters.py

```diff
@@ -0,0 +1,220 @@
+from sphinx.ext import autodoc
+
+from .mappers.python import (
+    PythonFunction,
+    PythonClass,
+    PythonMethod,
+    PythonData,
+    PythonAttribute,
+    PythonException,
+)
+from . import utils
+
+
+class AutoapiDocumenter(autodoc.Documenter):
+    def get_attr(self, obj, name, *defargs):
+        if (hasattr(self.env.app, 'registry')
+                and hasattr(self.env.app.registry, 'autodocattrgettrs')):
+            attrgetters = self.env.app.registry.autodoc_attrgettrs
+        else:
+            attrgetters = autodoc.AutoDirective._special_attrgetters
+
+        for type_, func in attrgetters.items():
+            if isinstance(obj, type_):
+                return func(obj, name, *defargs)
+
+        if name == '__doc__':
+            return obj.docstring
+
+        for child in obj.children:
+            if child.name == name:
+                return child
+
+        if defargs:
+            return defargs[0]
+
+        raise AttributeError(name)
```

Comment:
Does this get hit often? I wonder if it should have a bit more of a log message in order for users to debug this.